_source/logzio_collections/_prometheus-sources/dotnet-custom.md

```diff
@@ -0,0 +1,570 @@
+
+</div>
+
+</div>
+<!-- tab:end -->
+
+<!-- tab:start -->
+<div id="in-config">
+
+
+#### Send custom metrics to Logz.io with a Logz.io exporter defined by a config file
+
+**Before you begin, you'll need**: 
+
+* an application in .NET Core 3.1 and higher
+* an active account with Logz.io
+
+
+<div class="tasklist">
+
+
+##### Install the App.Metrics.Logzio package
+
+
+Install the App.Metrics.Logzio package from the Package Manager Console:
+
+```csharp
+Install-Package Logzio.App.Metrics
+```
+
+Alternatively, If you prefer to install the library manually, download the latest version from NuGet Gallery
+
+
+##### Create MetricsBuilder
+
+To create MetricsBuilder, copy and paste the following code into the function of the code that you need to export metris from:
+
+```csharp
+var metrics = new MetricsBuilder()
+                .Report.ToLogzioHttp("<<path_to_the_config_file>>")
+                .Build();
+```
+
+Add the following code to the configuration file:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+
+<Configuration>
+    <LogzioConnection>
+        <Endpoint> <<LISTENER-HOST>> </Endpoint>
+        <Token> <<METRICS-SHIPPING-TOKEN>> </Token>
+    </LogzioConnection>
+</Configuration>
+```
+
+{% include log-shipping/listener-var.html %} For HTTPS communication use port 8053. For HTTP communication use port 8052.
+
+{% include metric-shipping/replace-metrics-token.html %}
+
+
+##### Create Scheduler
+
+To create Scheduler, copy and paste the following code into the same function of the code as the MetricsBuilder:
+
+```csharp
+var scheduler = new AppMetricsTaskScheduler(
+                TimeSpan.FromSeconds(15),
+                async () => { await Task.WhenAll(metrics.ReportRunner.RunAllAsync()); });
+            scheduler.Start();
+```
+
+##### Add required metrics to your code
+
+You can send the following metrics from your code:
+
+* [Apdex (Application Performance Index)](https://www.app-metrics.io/getting-started/metric-types/apdex/)
+* [Counter](https://www.app-metrics.io/getting-started/metric-types/counters/)
+* [Gauge](https://www.app-metrics.io/getting-started/metric-types/gauges/)
+* [Histogram](https://www.app-metrics.io/getting-started/metric-types/histograms/)
+* [Meter](https://www.app-metrics.io/getting-started/metric-types/meters/)
+* [Timer](https://www.app-metrics.io/getting-started/metric-types/timers/)
+
+You must have at least one of the above metrics in your code to use the Logzio.App.Metrics. For example, to add a counter metric to your code, copy and paste the following code block into the same function of the code as the MetricsBuilder and Scheduler:
+
+```csharp
+var counter = new CounterOptions {Name = "my_counter", Tags = new MetricTags("test", "my_test")};
+            metrics.Measure.Counter.Increment(counter);
+
+            Thread.Sleep(10000);
+
+            metrics.Measure.Counter.Increment(counter);
+
+            Thread.Sleep(100000); 
+```
```

Comment:
Please delete the Thread.Sleep lines (they are not relevant), and make the lines to start from same position as var counter line (you can even put a new line after first line and then the 2 metrics.Measure lines one after another)